### PR TITLE
HDS-53 Added Webhook auth test (and refactored some test structure)

### PR DESCRIPTION
**Pull request by kingeric027**
1. Added a test for webhook authentication.
2. Moved the "integration" test from the demo.WebApi project into our tests project and renamed it OCUserAuthTests (it tests token authentication).
3. Created a TestService to store some common code as functions

### CODE REVIEW

File: demo/OrderCloud.DemoWebApi/Controllers/DemoController.cs [hunk 1]
```diff
@@ -41,24 +41,17 @@ public class DemoController : BaseController
...
+        public object HandleAddressSave([FromBody] WebhookPayloads.Addresses.Save payload)
+        {
+            return new
+            {
+                Action = "HandleAddressSave",
+                City = payload.Request.Body.City,
+                Foo = payload.ConfigData.Foo
+            };
+        }
+    }
```
Comment (crhistianramirez) [inline demo/OrderCloud.DemoWebApi/Controllers/DemoController.cs:39]: Can we add some comments? Also, the main use case of using disambiguate webhooks is when you want one route to handle multiple payloads. I think it'd be worth demonstrating at least another route (same route, different payload)

Comment (oliverheywood451) [reply demo/OrderCloud.DemoWebApi/Controllers/DemoController.cs:39]: I might also suggest we create a new file WebhookContoller.cs. Then if we want to show someone how to do webhooks we can link them to it and there's nothing else to confuse them.

File: demo/OrderCloud.DemoWebApi/AppSettings.cs [hunk 2]
```diff
@@ -8,5 +8,7 @@ namespace OrderCloud.DemoWebApi
 	public class AppSettings
 	{
 		public string MyDBConnectionString { get; set; }
-	}
+
+		public string WebhookHashKey { get; } = "myhashkey";
```
Comment (oliverheywood451) [inline demo/OrderCloud.DemoWebApi/AppSettings.cs]: You could have a comment on this property like "Should match the HashKey configured on your webhook in the Ordercloud portal" or something.

**Comment by oliverheywood451** [comment]
For tests, I'm a big fan of separating out OCUserAuthTests and calling them that instead of just Integration tests. 

I'm unsure about moving tests that call routes out of the Ordercloud.DemoWebApi.Tests project and into the Ordercloud.Catalyst.Tests project. What's the thinking?

In the original conception, there are two test projects. One for unit testing functionality inside the Ordercloud.Catalyst library. Another for integration tests against the Ordercloud.DemoWebApi. If somebody copied the demo folder, those tests would come with and form a template for them to continue writing tests. 

On the other hand, maybe two test projects is too complex? This question really goes down to big-picture project structure. Might be good to put heads together with the whole group.

**Review by crhistianramirez — APPROVED** [review]

**Pull request closed without merging**